openpype/plugins/actions/open_file_explorer.py

```diff
@@ -29,40 +29,30 @@ def process(self, session, **kwargs):
         project_name = session["AVALON_PROJECT"]
         asset_name = session.get("AVALON_ASSET", None)
         task_name = session.get("AVALON_TASK", None)
-        host_name = None    # never present in session
 
         project = get_project(project_name)
         asset = get_asset_by_name(project_name, asset_name)
 
-        data = self.get_workdir_data(project_doc=project,
-                                     asset_doc=asset,
-                                     task_name=task_name,
-                                     host_name=host_name)
+        data = get_template_data(project, asset, task_name)
+
         anatomy = Anatomy(project_name)
-        result = anatomy.format(data)
-
-        # todo: implement custom template keys instead of 'work'
-        # Get template key
-        # template_key = get_workfile_template_key_from_context(
-        #    asset_name, task_name, host_name, project_name, dbcon
-        # )
-        workdir = result["work"]["folder"]
-
-        # Keep only the part of the path that was formatted by splitting up to
-        # the first stub - we can only explore up to there
-        STUB = "<<NULL>>"
-        valid_dir = workdir.split(STUB, 1)[0]
-
-        # If the path endswith `/work` and the path does not exist but the
-        # parent does then we allow to go to the parent because it could for
-        # example be the hierarchical folder for `asset` in e.g. `asset/hero`
-        if not os.path.exists(valid_dir) and valid_dir.endswith("/work/"):
-            # /folder/ dirname is /folder so we rstrip the last / to be sure
-            parent_dir = os.path.dirname(valid_dir.rstrip("/"))
-            if os.path.exists(parent_dir):
-                valid_dir = parent_dir
-
-        path = os.path.normpath(valid_dir)
+        workdir = anatomy.templates_obj["work"]["folder"].format(data)
+
+        valid_workdir = workdir.split("{", 1)[0]
```

Comment:
This should have a comment that it's removing any potential unformatted parts of the path just for clarity sake.